File: external/HDF-EOS/hdfeos/samples/ReadFields.c
#include "hdf.h"


/*
 * In this example we will (1) open the "SwathFile" HDF file, (2) attach to
 * the "Swath1" swath, and (3) read data from the "Longitude" field.
 * 
 * Unlike the WriteField routine, we will read the field all at once.
 */


main()
{

    intn            status, i, j, k, start[2],stride[2],count[2];
    int32           swfid, SWid, attr[4];

    float32         lng[20][10];
    /* Allocate space for the longitude and spectral data */


    /*
     * Open the HDF swath file, "SwathFile.hdf".
     */

    swfid = SWopen("SwathFile.hdf", DFACC_READ);


    if (swfid != -1)
    {

	/*
	 * Attach the "Swath1" swath.
	 */

	SWid = SWattach(swfid, "Swath1");
	
	if (SWid != -1)
	{
	    /* Read the entire longitude field */
	    status = SWreadfield(SWid, "Longitude", NULL, NULL, NULL, lng);

	    /* Print field */
	    for (i = 0; i < 20; i++)
		for (j = 0; j < 10; j++)
		    printf("i j Longitude: %d %d %f\n", 
			   i, j, lng[i][j]);


	    /* Read User Attribute */
	    status = SWreadattr(SWid, "TestAttr", attr);
	    for (i=0;i<4;i++)
		printf("Attribute Entry %d: %d\n",i+1,attr[i]);

	}
    }
    status = SWdetach(SWid);
    status = SWclose(swfid);
    HEprint(stdout,0);
    
    return;
}

File: external/HDF-EOS/hdfeos/samples/readfields.f

c
c  In this example we will (1) open the "SwathFile" HDF file, (2) attach to
c  the "Swath1" swath, and (3) read data from the "Longitude" field.
c 


	program readfields

	integer            status, swrdfld, swrdattr, swdetach, swclose
	integer            start(2),stride(2),count(2), attr(4)
	integer*4          swfid, swid, swopen, swattach

	real*4         lng(10,20)
	
	integer DFACC_READ
	parameter (DFACC_READ=1)
	

c    
c     Open the HDF swath file, "SwathFile.hdf"
c 

	swfid = swopen("SwathFile.hdf", DFACC_READ)


	if (swfid .NE. -1) then

	   swid = swattach(swfid, "Swath1")

	   if (swid .NE. -1) then

	      
c     Read the entire Longitude field

	      start(1) = 0
	      start(2) = 0
	      stride(1) = 1
	      stride(2) = 1
	      count(1) = 10
	      count(2) = 20

	      status = swrdfld(swid, "Longitude", start, stride, count,
     1                         lng)

	      do i=1,20
		 do j=1,10
		    write(*,*)'i j Longitude ',i,j,lng(j,i)
		 enddo
	      enddo


c      Read Attribute	      
	      status = swrdattr(swid, "TestAttr", attr)
	      do i=1,4
		 write(*,*) 'Attribute Element', i, ':', attr(i)
	      enddo
	      
	      
	   endif
	endif
	
	status = swdetach(swid)
	status = swclose(swfid)

	stop
	end




